src/app/app.component.ts

```diff
@@ -1,9 +1,34 @@
-import { Component } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { NavigationEnd, Router } from '@angular/router';
+import { Subject } from 'rxjs';
+import { filter, takeUntil, tap } from 'rxjs/operators';
+
+import { EventsService } from 'src/app/modules/core/services/events.service';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
 })
-export class AppComponent {
+export class AppComponent implements OnInit, OnDestroy {
```

Comment:
Route events are one of the few things that we should be subscribing to in the app's top level component, which is perfect for tracking navigation events